### PR TITLE
coco/validate runtime between funfile and yml when build function

### LINKED ISSUE #734: Funfile 中  RUNTIME 和 template.yml 中不一致时给出警告
Funfile 中 RUNTIME 比 template.yml 要高，有用户在 custom runtime 的函数里使用了 python3 的 Runtime 进行 Funfile 安装碰到了问题。改成 custom 以后文件就解决了。对于这类问题建议给出警告。

### PR DESCRIPTION
**Pull request by coco-super**
fix #734 #738

### CODE REVIEW

File: lib/build/build.js [hunk 1]
```diff
@@ -153,6 +153,20 @@ async function recordMetaData(baseDir, functions, tplPath, metaPath, buildOps) {
...
+  const funfilePathContent = await file.readLines(funfilePath);
+
+  const runtimeLine = funfilePathContent.find(line => {
+    return line.indexOf('RUNTIME') !== -1;
+  });
+
+  if (!runtimeLine) { return; }
+
+  if (_.trim(_.trimStart(_.trim(runtimeLine), 'RUNTIME')) !== runtime) {
+    console.warn(red(`\nDetectionWarning: Fun has detected that the 'runtime' in your template.yml is inconsistent with that in Funfile.\n`));
```
Comment (vangie) [inline lib/build/build.js]: warn 用黄色或者橙色更好些。

Fun has detected that the 'runtime' in your template.yml is inconsistent with that in Funfile.

=>

The 'runtime' of 'servicexxx/funcationxxx' in your template.yml is inconsistent with that in Funfile.

File: lib/build/build.js [hunk 2]
```diff
@@ -153,6 +153,20 @@ async function recordMetaData(baseDir, functions, tplPath, metaPath, buildOps) {
...
+async function checkFunfileRuntime(funfilePath, runtime) {
+  const funfilePathContent = await file.readLines(funfilePath);
+
+  const runtimeLine = funfilePathContent.find(line => {
+    return line.indexOf('RUNTIME') !== -1;
+  });
+
+  if (!runtimeLine) { return; }
+
+  if (_.trim(_.trimStart(_.trim(runtimeLine), 'RUNTIME')) !== runtime) {
```
Comment (vangie) [inline lib/build/build.js]: 这里的嵌套太复杂了，改成正则表达式会不会好些？

File: lib/build/build.js [hunk 3]
```diff
@@ -153,6 +153,20 @@ async function recordMetaData(baseDir, functions, tplPath, metaPath, buildOps) {
   await recordMtimes([...metaPaths, tplPath], buildOps, metaPath);
 }
 
+async function checkFunfileRuntime(funfilePath, runtime) {
```
Comment (vangie) [inline lib/build/build.js]: checkRuntimeOfFunfile

File: lib/build/build.js (hunk 1 above)
Comment (coco-super) [reply lib/build/build.js]: done

File: lib/build/build.js (hunk 2 above)
Comment (coco-super) [reply lib/build/build.js]: 这段逻辑冗余了，已删除。

File: lib/build/build.js (hunk 3 above)
Comment (coco-super) [reply lib/build/build.js]: done

File: lib/build/parser.js [hunk 4]
```diff
@@ -144,9 +145,13 @@ If you have a requirement, you can submit the issue at https://github.com/alibab
         throw new Error('invalid RUNTIME for Funfile');
       }
 
-      const runtime = runtimeArgs[0].getValue();
+      const funfileRuntime = runtimeArgs[0].getValue();
```
Comment (vangie) [inline lib/build/parser.js]: runtimeOfFunfile 或者 rtOfFunfile

Comment (coco-super) [reply lib/build/parser.js]: done

**Pull request merged**